### ISSUE TITLE
IndexError during compute_residuals

### ISSUE THREAD

**Issue by romainmartinez** [CamDavidsonPilon/lifelines#612]
Hi Cam,

I successfully fitted a survival regression  (`CoxPHFitter`) on my data. However, I cannot compute the shoenfeld (`_compute_schoenfeld` called by `residuals = self.compute_residuals(training_df, kind="scaled_schoenfeld")`) and get the following error:

```python
df = pd.DataFrame(schoenfeld_residuals[E, :], columns=self.hazards_.columns, index=index[E])
IndexError: boolean index did not match indexed array along dimension 0; dimension is 3654 but corresponding boolean dimension is 3743
```

The error come from `schoenfeld_residuals[E, :]` with `schoenfeld_residuals` of shape `(3654, 6)` and `E` `(3743,)`.

I guess `E` correspond to the event and this a censorship issue?

This seems related to [this issue](https://stackoverflow.com/questions/48076384/lifelines-boolean-index-in-python-did-not-match-indexed-array-along-dimension-0).

By the way, thank you very much for lifelines–I didn’t know much on survival analysis and was ready in no time thank to the clear api and docs.

**Comment by CamDavidsonPilon** [comment]
Thanks for filing this. Can you answer:

1. Does `training_df` have a non-trivial index?
2. Are you using `strata` in the call to `fit`?

I'll try to reproduce the error locally now.

**Comment by CamDavidsonPilon** [comment]
Okay, I can reproduce the error locally, but it may be different than why you experienced it. Does your dataset have _only_ censored subjects for the minimum duration?

**Comment by CamDavidsonPilon** [comment]
AFAIK, this is fixed

**Comment by CamDavidsonPilon** [comment]
This fix is available on PyPI under 0.17.3

**Comment by romainmartinez** [comment]
Everything is working fine with the fix, thanks!
